src/vendor/components/navigation/TimePicker.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import MenuItem from '@material-ui/core/MenuItem';
+import TextField from '@material-ui/core/TextField';
+import { ArrayWrapper, selectFrom } from '../../vectors';
+import { missing } from '../../utils';
+import Date from '../../dates';
+
+const styles = () => ({
+  root: {
+    display: 'inline-block',
+    paddingTop: '2rem',
+    margin: '0 20px',
+  },
+});
+
+class TimePicker extends Component {
+  constructor(props) {
+    super(props);
+
+    const { value } = this.props;
+
+    this.state = { value };
+  }
+
+  render() {
+    const { classes, id, label, handleChange, options } = this.props;
+    const { value } = this.state;
+
+    return (
+      <form className={classes.root} autoComplete="off">
+        <TextField
+          select
+          name={id}
+          label={label}
+          value={value}
+          onChange={handleChange}>
+          {options.map(({ label, id }) => (
+            <MenuItem key={id} value={id}>
+              {label}
+            </MenuItem>
+          ))}
+        </TextField>
+      </form>
+    );
+  }
+}
+
+/*
+ENSURE THE options ARE UPDATED
+RETURN CORRECTED VALUE
+ */
+TimePicker.prepare = props => {
+  const { options, value, defaultValue } = props;
+
+  if (missing(value)) return defaultValue;
+
+  if (
+    selectFrom(options)
+      .select('id')
+      .includes(value)
+  )
+    return value;
+
+  options.push({
+    id: value,
+    label: Date.newInstance(value).format('MMM dd, yyyy'),
+  });
```

Comment:
Above:  The TimePicker is different than the other pickers. Other pickers will confirm the URL has a legit value, and if not then use the default value. For the TimePicker; if you put an ending date in the URL, it will be used as an option.